### ISSUE TITLE
False positive error message regarding "subscript for class" appears transiently

### ISSUE THREAD

**Issue by sgtmjrme** [microsoft/pyright#3944]
## Environment data

-   Language Server version: v2022.9.10
-   OS and version: Ubuntu 20.04
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10

## Code Snippet
The following code snippet triggers a warning in Pylance "Subscript for class "list" will generate runtime exception; enclose type annotation in quotes"

```python
class testclass:
    class baseclass:
        pass

something: 'list[testclass.baseclass]' = [] <<< Pylance warning
```

## Expected behavior

The above code should not throw an error - the list _is_ enclosed in quotes, and runs perfectly fine without issues.  Pylance seems to throw the error if the class in the list is dotted (as if it was a class in a class or the class is in another file).  

## Actual behavior

Pylance generates the warning.  

## Logs

Not sure what part of the log is necessary, I have a lot of code open so it was throwing a lot of data at me.  Think I grabbed the test code section.  

```
(160482) [FG] parsing: /nfs/scripts/dev/rjjohnso/test.py (19ms)
(160482) [FG] parsing: /root/.vscode-server/extensions/ms-python.vscode-pylance-2022.9.10/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 0ms] (126ms)
(160482) [FG] binding: /root/.vscode-server/extensions/ms-python.vscode-pylance-2022.9.10/dist/typeshed-fallback/stdlib/builtins.pyi (35ms)
(160482) [FG] binding: /nfs/scripts/dev/rjjohnso/test.py (0ms)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.271, which I just published. It will also be included in a future release of pylance.
